internal/app/init/main.go

```diff
@@ -287,54 +307,99 @@ func startKubernetesServices(data *userdata.UserData) {
 	)
 }
 
+func sync() {
+	syncdone := make(chan struct{})
+
+	go func() {
+		defer close(syncdone)
+		unix.Sync()
+	}()
+
+	log.Printf("Waiting for sync...")
+
+	for i := 29; i >= 0; i-- {
+		select {
+		case <-syncdone:
+			log.Printf("Sync done")
+			return
+		case <-time.After(time.Second):
+		}
+		if i != 0 {
+			log.Printf("Waiting %d more seconds for sync to finish", i)
+		}
+	}
+
+	log.Printf("Sync hasn't completed in time, aborting...")
+}
+
+func reboot() {
+	// See http://man7.org/linux/man-pages/man2/reboot.2.html.
+	sync()
+
+	// nolint: errcheck
+	unix.Reboot(rebootFlag)
+
+	if *inContainer {
```

Comment:
Should these be a gate at the beginning of this function? I think we don't want to issue any commands in container mode.

Reply:
I'm open, I thought that if we run more code in container mode, we expose it for more testing, even if it's no-op in container mode. but I can return earlier of course

Reply:
I'm just thinking that reboot will actually reboot the host running the talos container, and we may not want to do that?

Reply:
Hmm... maybe we should `exit 0` here?

Reply:
Oh are you saying `unix.Reboot(rebootFlag)` is a no-op on container mode?

Reply:
so, not sure if it's no-op purely, but man page says:

```
   Behavior inside PID namespaces
       Since Linux 3.4, if reboot() is called from a PID namespace other
       than the initial PID namespace with one of the cmd values listed
       below, it performs a "reboot" of that namespace: the "init" process
       of the PID namespace is immediately terminated, with the effects
       described
```

So looks like it terminates `init`, so effectively kills this process and container with it. So it doesn't do any harm.